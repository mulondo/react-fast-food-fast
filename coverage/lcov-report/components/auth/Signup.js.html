<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/auth/Signup.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/auth</a> Signup.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {signUp} from '../../actions/signUpAction';
&nbsp;
// import food from '../../images/food.jpg';
&nbsp;
export class Register extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email:'',
      username:'',
      telephone:'',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
&nbsp;
  onChange(e){
    return this.setState({ [e.target.name]: e.target.value });
  }
&nbsp;
  
  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      telephone:this.state.telephone
    };
    // this.props.signUp(newUser);
    signUp(newUser);
    
  }
&nbsp;
&nbsp;
  render() {
    let userInformation = (
      &lt;div className="signupmodal"&gt;
          &lt;form className="modal-content animate" name="sigin" onSubmit={this.onSubmit}&gt;
              &lt;h1&gt;Fast-Food-Fast&lt;/h1&gt;
              &lt;div className="imgcontainer"&gt;    
                  {/* &lt;img src={food} alt="User" className="food"/&gt; */}
              &lt;/div&gt;
              &lt;div className="container"&gt;
&nbsp;
                    &lt;label forhtml="username"&gt;&lt;b&gt;Username&lt;/b&gt;&lt;/label&gt;
                    &lt;input type="text"
                      placeholder="Enter Username" 
                      value={this.state.username} 
                      onChange={this.onChange} 
                      name="username" 
                      required/&gt;
&nbsp;
                    &lt;label forhtml="Email"&gt;&lt;b&gt;Email&lt;/b&gt;&lt;/label&gt;
                    &lt;input 
                      type="email" 
                      placeholder="example@gmail.com" 
                      name="email" 
                      value={this.state.email}
                      onChange={this.onChange} 
                      required/&gt;
&nbsp;
                    &lt;label forhtml="telephone"&gt;&lt;b&gt;Phone number&lt;/b&gt;&lt;/label&gt;
                    &lt;input 
                      type="text" 
                      id="telephone" 
                      placeholder="+256-703-492035" 
                      pattern="\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                      value={this.state.telephone}
                      onChange={this.onChange} 
                      name="telephone" 
                      required/&gt;
&nbsp;
                    &lt;label forhtml="password"&gt;&lt;b&gt;Password&lt;/b&gt;&lt;/label&gt;
                    &lt;input 
                      type="password" 
                      id="pwd"
                      value={this.state.password}
                      onChange={this.onChange}
                      placeholder="***************"
                      name="password" 
                      required/&gt;
&nbsp;
                      &lt;input type="submit" className="signupbtn" value="Register"/&gt;
              &lt;/div&gt;
        &lt;/form&gt;
    &lt;/div&gt;
      
    );
    return &lt;div className="flex-grow-1"&gt;{userInformation}&lt;/div&gt;;
  }
}
&nbsp;
// Register.prototype={
//   signUp:PropTypes.func.isRequired
// }
&nbsp;
export default connect(null, {signUp})(Register);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Feb 12 2019 22:43:33 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
