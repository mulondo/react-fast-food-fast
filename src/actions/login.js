const API_HOST_URL = process.env.API_URL;

export const Login = (userdata, history) => dispatch => {
  return fetch(`${API_HOST_URL}/auth/login`, {
    method: 'POST',
    body: JSON.stringify(userdata),
    headers: {
      'content-type': 'application/json',
    }
  })
    .then(res => res.json())
    .then(responseData => {
      if (responseData.token){
        alert(responseData.message);
        if(responseData.message === 'You have been successfully logged in!'){
          history.push('/menu');
          localStorage.setItem('usertoken');
        }
      } 
      else {
        alert(responseData.error)
      }
    });
};

export function logout() {
  return function(dispatch) {
    localStorage.removeItem('usertoken');
    dispatch({ type: SIGNOUT });
  };
 }
