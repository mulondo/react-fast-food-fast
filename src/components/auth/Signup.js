import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {signUp} from '../../actions/signUpAction';

// import food from '../../images/food.jpg';

export class Register extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email:'',
      username:'',
      telephone:'',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e){
    return this.setState({ [e.target.name]: e.target.value });
  }

  
  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      telephone:this.state.telephone
    };
    this.props.signUp(newUser);
    
  }

  render() {
    let userInformation = (
      <div className="signupmodal">
          <form className="modal-content animate" name="sigin" onSubmit={this.onSubmit}>
              <h1>Fast-Food-Fast</h1>
              <div className="imgcontainer">    
                  {/* <img src={food} alt="User" className="food"/> */}
              </div>
              <div className="container">

                    <label forhtml="username"><b>Username</b></label>
                    <input type="text"
                      placeholder="Enter Username" 
                      value={this.state.username} 
                      onChange={this.onChange} 
                      name="username" 
                      required/>

                    <label forhtml="Email"><b>Email</b></label>
                    <input 
                      type="email" 
                      placeholder="example@gmail.com" 
                      name="email" 
                      value={this.state.email}
                      onChange={this.onChange} 
                      required/>

                    <label forhtml="telephone"><b>Phone number</b></label>
                    <input 
                      type="text" 
                      id="telephone" 
                      placeholder="+256-703-492035" 
                      pattern="\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                      value={this.state.telephone}
                      onChange={this.onChange} 
                      name="telephone" 
                      required/>

                    <label forhtml="password"><b>Password</b></label>
                    <input 
                      type="password" 
                      id="pwd"
                      value={this.state.password}
                      onChange={this.onChange}
                      placeholder="***************"
                      name="password" 
                      required/>

                      <input type="submit" className="signupbtn" value="Register"/>
              </div>
        </form>
    </div>
      
    );
    return <div className="flex-grow-1">{userInformation}</div>;
  }
}

// Register.prototype={
//   signUp:PropTypes.func.isRequired
// }

export default connect(null, {signUp})(Register);