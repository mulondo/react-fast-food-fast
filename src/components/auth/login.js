import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {Login} from '../../actions/login';

// import food from '../../images/food.jpg';

export class Userlogin extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username:'',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e){
    return this.setState({ [e.target.name]: e.target.value });
  }
  
  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.Login(newUser, this.props.history);  
  }

  render() {
    let userInformation = (
      <div className="signupmodal">
          <form className="modal-content animate" name="sigin" onSubmit={this.onSubmit}>
              <h1>Fast-Food-Fast</h1>
              <div className="imgcontainer">    
                  {/* <img src={food} alt="User" className="food"/> */}
              </div>
              <div className="container">
                    <label forhtml="username"><b>Username</b></label>
                    <input type="text"
                      placeholder="Enter Username" 
                      value={this.state.username} 
                      onChange={this.onChange} 
                      name="username" 
                      required/>

                    <label forhtml="password"><b>Password</b></label>
                    <input 
                      type="password" 
                      id="pwd"
                      value={this.state.password}
                      onChange={this.onChange}
                      placeholder="***************"
                      name="password" 
                      required/>
                      <input type="submit" className="signupbtn" value="Login"/>
                      <Link to="/signup">
                      <h4>Sign up</h4>
                      </Link>
              </div>
        </form>
    </div>
      
    );
    return <div className="flex-grow-1">{userInformation}</div>;
  }
}


export default connect(null, {Login})(Userlogin);