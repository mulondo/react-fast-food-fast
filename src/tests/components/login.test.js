import React from 'react';
import { shallow, mount } from 'enzyme';
import { Userlogin } from '../../components/auth/login';


describe('<Userlogin/>', () => {
  const editor = shallow(<Userlogin registerUser={jest.fn()} />);
  const props = {
    registerUser: (user) => jest.fn(),
    Login: jest.fn()
  };

  it('should render without crashing', () => {
    expect(editor).toMatchSnapshot();
  });



it('renders a username input', () => {
  expect(shallow(<Userlogin />).find('#username').length).toEqual(0);
});

it('renders a password input', () => {
  expect(shallow(<Userlogin />).find('#password').length).toEqual(0);
});

it('test Login component onchange', () => {
  const wrapper = shallow(<Userlogin {...props} />);
  const instance = wrapper.instance();
  instance.onChange({ target: { name: 'name', value: 'value' } });
  const e = { preventDefault: () => {} };
  instance.onSubmit(e);
  expect(instance.state.name).toEqual('value');
});

});