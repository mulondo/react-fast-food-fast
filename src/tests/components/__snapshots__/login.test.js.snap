// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Userlogin/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Userlogin
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="signupmodal"
      >
        <form
          className="modal-content animate"
          name="sigin"
          onSubmit={[Function]}
        >
          <h1>
            Fast-Food-Fast
          </h1>
          <div
            className="imgcontainer"
          />
          <div
            className="container"
          >
            <label
              forhtml="username"
            >
              <b>
                Username
              </b>
            </label>
            <input
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="password"
            >
              <b>
                Password
              </b>
            </label>
            <input
              id="pwd"
              name="password"
              onChange={[Function]}
              placeholder="***************"
              required={true}
              type="password"
              value=""
            />
            <input
              className="signupbtn"
              type="submit"
              value="Login"
            />
            <Link
              replace={false}
              to="/signup"
            >
              <h4>
                Sign up
              </h4>
            </Link>
          </div>
        </form>
      </div>,
      "className": "flex-grow-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="modal-content animate"
          name="sigin"
          onSubmit={[Function]}
        >
          <h1>
            Fast-Food-Fast
          </h1>
          <div
            className="imgcontainer"
          />
          <div
            className="container"
          >
            <label
              forhtml="username"
            >
              <b>
                Username
              </b>
            </label>
            <input
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="password"
            >
              <b>
                Password
              </b>
            </label>
            <input
              id="pwd"
              name="password"
              onChange={[Function]}
              placeholder="***************"
              required={true}
              type="password"
              value=""
            />
            <input
              className="signupbtn"
              type="submit"
              value="Login"
            />
            <Link
              replace={false}
              to="/signup"
            >
              <h4>
                Sign up
              </h4>
            </Link>
          </div>
        </form>,
        "className": "signupmodal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Fast-Food-Fast
            </h1>,
            <div
              className="imgcontainer"
            />,
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Login"
              />
              <Link
                replace={false}
                to="/signup"
              >
                <h4>
                  Sign up
                </h4>
              </Link>
            </div>,
          ],
          "className": "modal-content animate",
          "name": "sigin",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fast-Food-Fast",
            },
            "ref": null,
            "rendered": "Fast-Food-Fast",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "imgcontainer",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  forhtml="username"
                >
                  <b>
                    Username
                  </b>
                </label>,
                <input
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter Username"
                  required={true}
                  type="text"
                  value=""
                />,
                <label
                  forhtml="password"
                >
                  <b>
                    Password
                  </b>
                </label>,
                <input
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="***************"
                  required={true}
                  type="password"
                  value=""
                />,
                <input
                  className="signupbtn"
                  type="submit"
                  value="Login"
                />,
                <Link
                  replace={false}
                  to="/signup"
                >
                  <h4>
                    Sign up
                  </h4>
                </Link>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Username
                  </b>,
                  "forhtml": "username",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Username",
                  },
                  "ref": null,
                  "rendered": "Username",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "username",
                  "onChange": [Function],
                  "placeholder": "Enter Username",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Password
                  </b>,
                  "forhtml": "password",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "pwd",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "***************",
                  "required": true,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signupbtn",
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <h4>
                    Sign up
                  </h4>,
                  "replace": false,
                  "to": "/signup",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "h4",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signupmodal"
        >
          <form
            className="modal-content animate"
            name="sigin"
            onSubmit={[Function]}
          >
            <h1>
              Fast-Food-Fast
            </h1>
            <div
              className="imgcontainer"
            />
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Login"
              />
              <Link
                replace={false}
                to="/signup"
              >
                <h4>
                  Sign up
                </h4>
              </Link>
            </div>
          </form>
        </div>,
        "className": "flex-grow-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="modal-content animate"
            name="sigin"
            onSubmit={[Function]}
          >
            <h1>
              Fast-Food-Fast
            </h1>
            <div
              className="imgcontainer"
            />
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Login"
              />
              <Link
                replace={false}
                to="/signup"
              >
                <h4>
                  Sign up
                </h4>
              </Link>
            </div>
          </form>,
          "className": "signupmodal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Fast-Food-Fast
              </h1>,
              <div
                className="imgcontainer"
              />,
              <div
                className="container"
              >
                <label
                  forhtml="username"
                >
                  <b>
                    Username
                  </b>
                </label>
                <input
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter Username"
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  forhtml="password"
                >
                  <b>
                    Password
                  </b>
                </label>
                <input
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="***************"
                  required={true}
                  type="password"
                  value=""
                />
                <input
                  className="signupbtn"
                  type="submit"
                  value="Login"
                />
                <Link
                  replace={false}
                  to="/signup"
                >
                  <h4>
                    Sign up
                  </h4>
                </Link>
              </div>,
            ],
            "className": "modal-content animate",
            "name": "sigin",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fast-Food-Fast",
              },
              "ref": null,
              "rendered": "Fast-Food-Fast",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "imgcontainer",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    forhtml="username"
                  >
                    <b>
                      Username
                    </b>
                  </label>,
                  <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter Username"
                    required={true}
                    type="text"
                    value=""
                  />,
                  <label
                    forhtml="password"
                  >
                    <b>
                      Password
                    </b>
                  </label>,
                  <input
                    id="pwd"
                    name="password"
                    onChange={[Function]}
                    placeholder="***************"
                    required={true}
                    type="password"
                    value=""
                  />,
                  <input
                    className="signupbtn"
                    type="submit"
                    value="Login"
                  />,
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    <h4>
                      Sign up
                    </h4>
                  </Link>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Username
                    </b>,
                    "forhtml": "username",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Username",
                    },
                    "ref": null,
                    "rendered": "Username",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "username",
                    "onChange": [Function],
                    "placeholder": "Enter Username",
                    "required": true,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Password
                    </b>,
                    "forhtml": "password",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "pwd",
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "***************",
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signupbtn",
                    "type": "submit",
                    "value": "Login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <h4>
                      Sign up
                    </h4>,
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sign up",
                    },
                    "ref": null,
                    "rendered": "Sign up",
                    "type": "h4",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
